repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/PreProcessorUtils.java

```diff
@@ -198,4 +212,208 @@ public static void verifyDuplicateAssetByName(String typeName, String assetName,
             throw new AtlasBaseException(AtlasErrorCode.BAD_REQUEST, errorMessage);
         }
     }
+
+    public static void isValidLexoRank(String input, String glossaryQualifiedName, String parentQualifiedName, EntityDiscoveryService discovery) throws AtlasBaseException {
+        String pattern = "^0\\|[0-9a-z]{6}:(?:[0-9a-z]{0," + LEXORANK_HARD_LIMIT + "})?$";
+        // TODO : To remove this after migration is successful on all tenants and custom-sort is successfully GA
+        Boolean requestFromMigration = RequestContext.get().getRequestContextHeaders().getOrDefault("x-atlan-request-id", "").contains("custom-sort-migration");
+        Pattern regex = Pattern.compile(pattern);
+
+        Matcher matcher = regex.matcher(input);
+
+        if(!matcher.matches()){
+            throw new AtlasBaseException("Invalid LexicographicSortOrder");
+        }
+        if(!requestFromMigration) {
+            Map<String, Object> dslQuery = createDSLforCheckingPreExistingLexoRank(input, glossaryQualifiedName, parentQualifiedName);
+            List<AtlasEntityHeader> categories = new ArrayList<>();
+            try {
+                IndexSearchParams searchParams = new IndexSearchParams();
+                searchParams.setAttributes(new HashSet<>());
+                searchParams.setDsl(dslQuery);
+                categories = discovery.directIndexSearch(searchParams).getEntities();
+            } catch (AtlasBaseException e) {
+                e.printStackTrace();
```

Comment:
Sent an exception here, in case indexsearch fails for some reason we should not allow further processing of the request, add a error log statement along with the exception